classes/Presenter/Transaction/TransactionPresenter.php

```diff
@@ -0,0 +1,106 @@
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2019 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+namespace PrestaShop\Module\PrestashopCheckout\Presenter\Transaction;
+
+use PrestaShop\Module\PrestashopCheckout\Adapter\LinkAdapter;
+use PrestaShop\Module\PrestashopCheckout\Presenter\PresenterInterface;
+
+/**
+ * Present the pending orders for the reporting
+ */
+class TransactionPresenter implements PresenterInterface
+{
+    /**
+     * present pending orders
+     *
+     * @return array
+     */
+    public function present()
+    {
+        $link = new LinkAdapter();
+        $transactions = $this->getTransactions();
+
+        foreach ($transactions as &$transaction) {
+            $userInfos = $this->getUserInfos($transaction['order_reference']);
+            $transaction['transactionID'] = $transaction['transaction_id'];
+            $transaction['order_id'] = $this->getOrderIDByOrderReference($transaction['order_reference']);
+            $transaction['orderLink'] = $link->getAdminLink('AdminOrders', true, [], ['id_order' => $transaction['order_id'], 'vieworder' => 1]);
+            $transaction['username'] = $userInfos['name'];
+            $transaction['userProfileLink'] = $link->getAdminLink('AdminCustomers', true, [], ['id_customer' => $userInfos['userID'], 'viewcustomer' => 1]);
+            $currency = new \Currency($transaction['id_currency']);
+            $transaction['before_commission'] = \Tools::displayPrice($transaction['amount'], $currency);
+            $transaction['type'] = strpos($transaction['amount'], '-') !== false ? 'Refund' : 'Payment';
+            $transaction['commission'] = '-';
+            $transaction['total_paid'] = '-';
+        }
+
+        return $transactions;
+    }
+
+    private function getTransactions()
+    {
+        $sql = 'SELECT * FROM `' . _DB_PREFIX_ . 'order_payment` o
+            WHERE payment_method = "Prestashop Checkout"
+            ORDER BY date_add DESC
+            LIMIT 1000
+        ';
+
+        return \Db::getInstance()->executeS($sql);
+    }
+
+    private function getOrderIDByOrderReference($reference)
+    {
+        $sql = 'SELECT id_order FROM `' . _DB_PREFIX_ . 'orders` o
+            WHERE reference = "' . $reference . '"
+        ';
+
+        return \Db::getInstance()->getValue($sql);
+    }
+
+    /**
+     * getUserInfos
+     *
+     * @param string $orderReference
+     *
+     * @return string
+     */
+    private function getUserInfos($orderReference)
+    {
+        $link = new LinkAdapter();
+        $sql = 'SELECT id_customer FROM `' . _DB_PREFIX_ . 'orders` WHERE reference = "' . $orderReference . '"';
+        $userID = \Db::getInstance()->getRow($sql);
+
+        $sql = 'SELECT firstname,lastname FROM `' . _DB_PREFIX_ . 'customer` WHERE id_customer = ' . (int) $userID['id_customer'];
```

Comment:
Instantiate customer ? I don't know if a direct query to the Database is faster than instantiate an object but i Think it is more readable.

Reply:
in order to instantiate the object you need the database infos, and like doctrine the prestashop object model will load every props of an object so it might be slightly more readable but at some perf cost that IMO is not enough since here we have a very small query.
@apacios @Matt75 wdyt ?

Reply:
As it is a simple query it seems good to me like this. It's readable and it's a realy quick query.